File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/jodel/feed/JodelFeedViewModel.java
package de.jbamberger.offlineclient.ui.jodel.feed;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import java.util.List;

import javax.inject.Inject;

import de.jbamberger.offlineclient.source.jodel.JodelRepository;
import de.jbamberger.offlineclient.source.jodel.model.Post;
import de.jbamberger.offlineclient.util.Resource;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

public class JodelFeedViewModel extends ViewModel {

    private final LiveData<Resource<List<Post>>> posts;
    private JodelRepository repository;

    @Inject
    public JodelFeedViewModel(JodelRepository repository) {
        this.repository = repository;
        posts = repository.getPosts();
    }

    public LiveData<Resource<List<Post>>> getPosts() {
        return posts;
    }

    public void refresh() {
        repository.getPosts();
    }
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/jodel/feed/JodelFeedViewModel.kt
package de.jbamberger.offlineclient.ui.jodel.feed

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel
import de.jbamberger.api.Resource
import de.jbamberger.api.jodel.JodelRepository
import de.jbamberger.api.jodel.model.Post
import javax.inject.Inject

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

class JodelFeedViewModel @Inject
constructor(private val repository: JodelRepository) : ViewModel() {

    val posts: LiveData<Resource<List<Post>>> = repository.getPosts()

    fun refresh() {
        repository.getPosts()
    }
}
